static/app/components/button.stories.tsx

```diff
@@ -26,37 +18,38 @@ export default storyBook('Button', story => {
     );
   });
 
-  const propMatrix = {
+  const propMatrix: PropMatrix<typeof Button> = {
     borderless: [false, true],
     busy: [false, true],
     children: ['Save', undefined],
     icon: [undefined, <IconDelete key="" />],
-    priority: priorities,
-    size: sizes,
+    priority: ['default', 'primary', 'danger', 'link'],
+    size: ['md', 'sm', 'xs', 'zero'],
     disabled: [false, true],
     external: [false, true],
     title: [undefined, 'Save Now'],
     translucentBorder: [false, true],
   };
+
   story('Props', () => (
     <div>
-      <Matrix
-        component={Button}
+      <Matrix<typeof Button>
```

Comment:
It should be able to infer it from `Render` no?

Reply:
maybe in the simple cases it can be inferred, but so many of the stories i've written are using an anonymous function that doesn't explicitly say it's props

ie: `render={(props) => ...splat them out... }`

also, the Tabs example uses a combination of `typeof Tab & typeof TabItem` which worked out really well i think.

auto-inferring would be cool though because the syntax is clunky. happy to iterate now or later on it